editor/filesystem_dock.cpp

```diff
@@ -2647,8 +2660,79 @@ void FileSystemDock::_file_multi_selected(int p_index, bool p_selected) {
 	call_deferred(SNAME("_update_import_dock"));
 }
 
+void FileSystemDock::_tree_mouse_exited() {
+	if (holding_branch) {
+		_reselect_items_selected_on_drag_begin();
+	}
+}
+
+void FileSystemDock::_reselect_items_selected_on_drag_begin(bool reset) {
+	TreeItem *selected_item = tree->get_next_selected(tree->get_root());
+	if (selected_item) {
+		selected_item->deselect(0);
+	}
+	if (!tree_items_selected_on_drag_begin.is_empty()) {
+		bool reselected = false;
+		for (TreeItem *item : tree_items_selected_on_drag_begin) {
+			if (item->get_tree()) {
+				item->select(0);
+				reselected = true;
+			}
+		}
+
+		if (reset) {
+			tree_items_selected_on_drag_begin.clear();
+		}
+
+		if (!reselected) {
+			// If couldn't reselect the items selected on drag begin, select the "res://" item.
+			tree->get_root()->get_child(1)->select(0);
+		}
+	}
+
+	files->deselect_all();
+	if (!list_items_selected_on_drag_begin.is_empty()) {
+		for (const int idx : list_items_selected_on_drag_begin) {
+			files->select(idx, false);
+		}
+
+		if (reset) {
+			list_items_selected_on_drag_begin.clear();
+		}
+	}
+}
+
 void FileSystemDock::_tree_gui_input(Ref<InputEvent> p_event) {
 	Ref<InputEventKey> key = p_event;
+
+	Ref<InputEventMouseMotion> mm = p_event;
+	if (mm.is_valid()) {
+		TreeItem *item = tree->get_item_at_position(mm->get_position());
+		if (item && holding_branch) {
+			String fpath = item->get_metadata(0);
+			while (!fpath.ends_with("/") && fpath != "res://" && item->get_parent()) { // Find the parent folder tree item.
+				item = item->get_parent();
+				fpath = item->get_metadata(0);
+			}
+
+			TreeItem *deselect_item = tree->get_next_selected(tree->get_root());
+			while (deselect_item) {
+				deselect_item->deselect(0);
+				deselect_item = tree->get_next_selected(deselect_item);
+			}
```

Comment:
Couldn't it be replaced by `tree->deselect_all()`?

Reply:
```cpp
TreeItem *item = get_next_selected(get_root());
while (item) {
	item->deselect(selected_col);
	TreeItem *prev_item = item;
	item = get_next_selected(get_root());
	ERR_FAIL_COND(item == prev_item);
}
```
`deselect_all()` causes error for some reason when multiple files were selected and then get deselected. Idk if it is because it uses `selected_col`  or always passes root as argument to `get_next_selected()`.